docs/design/fabric-v2+/fpc-cc-invocation.puml

```diff
@@ -141,35 +146,35 @@ group invoke
       Enclave1 -> Enclave1 : ccrsp_msg.enclave_id <- Enclave_Id
       return <ccrsp_msg>
 
-    alt lightweight FPC
-      ccrsp_msg.rw_set <- encrypted_rw_set
-      note right
-        with chaincode-hosted validation, the stub will keep track of the 
-        R/W set (of cleartext keys but encrypted values) and pass it on to
-        the validation transaction as part of the response message.
-      end note
+     alt FPC "light"
+       ECC1 <-> ECC1 : ccrsp_msg.rw_set <- encrypted_rw_set
+       note right ECC1
+         with chaincode-hosted validation, the stub will keep track of the 
+         R/W set (of cleartext keys but encrypted values) and pass it on to
+         the validation transaction as part of the response message.
+       end note
     end
 
     return <ccrsp_msg>
 
-    alt  MVP
-      Peer -> Peer : create proposal response <args, read/writeset, result=<result_enc, Sig_Enclave, Enclave_VK>, ...>
+  alt  MVP
+    Peer -> Peer : create proposal response <args, read/writeset, result=<result_enc, Sig_Enclave, Enclave_VK>, ...>
       note right Peer
           Note that the result field of the proposal response contains
           a message that comprises the encrypted result produced by the
           enclave, the enclave signature, and the corresponding
           verification key.
       end note
 
-    else Post-MVP: Validation
-      Peer -> Peer : create proposal response <args, read/writeset, result=result_enc, ...>
-      Peer -> Peer : Replace endorsement signature & cert of peer with Sig_enclave & CC_Params.Certificate
+  else Post-MVP: Validation
+    Peer -> Peer : create proposal response <args, read/writeset, result=result_enc, ...>
+    Peer -> Peer : Replace endorsement signature & cert of peer with Sig_enclave & CC_Params.Certificate
       note right Peer
           This step is performed using an FPC-specific endorsement plugin.
       end note
-    end
+  end
 
-    return proposal response
+  return proposal response
 end
 
 group complete transaction
```

Comment:
a note, or alt should be added

Reply:
i added the validation transactions inside the validation.uml, where it would seem more natural. But it is true the handover between cc-invocation and fpc-validation has to be fixed. It essentially means referring to enough enclave endorsements and moving the submit transaction to validation